src/main/kotlin/registration/RegistrationStrategyImpl.kt

```diff
@@ -0,0 +1,69 @@
+package registration
+
+import com.github.kittinunf.result.Result
+import main.ConfigKeys
+import notary.CONFIG
+import notary.IrohaCommand
+import notary.IrohaOrderedBatch
+import notary.IrohaTransaction
+import sidechain.iroha.consumer.IrohaConsumer
+import sidechain.iroha.consumer.IrohaConverterImpl
+import sidechain.iroha.consumer.IrohaNetworkImpl
+
+/**
+ * Effective implementation of [RegistrationStrategy]
+ */
+class RegistrationStrategyImpl(
+    val ethFreeWalletsProvider: EthFreeWalletsProvider,
+    val irohaConsumer: IrohaConsumer
+) : RegistrationStrategy {
+
+    /**
+     * Register new notary client
+     * - CreateAccount with client name
+     * - SetAccountDetail on client account with assigned relay wallet from notary pool of free relay addresses
+     * - SetAccountDetail on notary node account to mark relay address in pool as assigned to the particular user
+     * @param name - client name
+     * @param pubkey - client public key
+     */
+    override fun register(name: String, pubkey: String): Result<Unit, Exception> {
```

Comment:
Please check what happens if stateful validation failed for one of iroha transactions.

Reply:
done